### PR TITLE
docs/library/struct: Embed format tables.

### PR DESCRIPTION
**Pull request by laurensvalk**
This makes it easier to see what is supported.

It also provides a place to comment about long support, which wasn't included before.

Signed-off-by: Laurens Valk <laurens@pybricks.com>

### CODE REVIEW

**Comment by mattytrentini** [comment]
I think this is _much_ more readable! Nice work @laurensvalk.

---
Here's a quick render of before and after so it's easier for folks can see:

image

---

And after (though it includes my footnote suggestion):

image

File: docs/library/struct.rst [hunk 1]
```diff
@@ -6,11 +6,56 @@
...
++--------+--------------------+-------------+---------------+
+| d      | double             | float (2)   | 8             |
++--------+--------------------+-------------+---------------+
+| s      | char[]             | bytes       |               |
++--------+--------------------+-------------+---------------+
+| P      | void *             | integer     |               |
++--------+--------------------+-------------+---------------+
+
+   - \(1\) Requires long support when used with values larger than 30 bits.
+   - \(2\) Requires floating point support.
```
[the remark this answers is not in the document]
Comment (laurensvalk) [reply docs/library/struct.rst:58]: Good idea, thanks!

[the remark this answers is not in the document]
Comment (dpgeorge) [reply docs/library/struct.rst:58]: Is it correct to use this footnote notation?  In the rendered example below given by @mattytrentini , it has `[1] (1,2,3,4): ...` and `[2]: (1,2): ...` which seems quite confusing, what do those tuples mean?

[the remark this answers is not in the document]
Comment (laurensvalk) [reply docs/library/struct.rst:58]: I'll have a look at how to suppress those.

[the remark this answers is not in the document]
Comment (dpgeorge) [reply docs/library/struct.rst]: OK, that's reasonable.  It has a link to the footnote so that's good.

[the remark this answers is not in the document]
Comment (dlech) [reply docs/library/struct.rst]: >it has [1] (1,2,3,4): ... and [2]: (1,2): ... which seems quite confusing, what do those tuples mean?

The "tuple" contains reverse links back to where the footnote is referenced (so footnote 1 is referenced 4 times and 2 is referenced 2 times). I suppose it is useful for finding all references to a footnote.

[the remark this answers is not in the document]
Comment (laurensvalk) [reply docs/library/struct.rst]: It seemed more confusing than helpful for this use case (and I couldn't suppress the tuple), so I left it out.

**Pull request merged**